### PR TITLE
ST: basic OLM tests

### PR DESCRIPTION
**Pull request by Frawless**
### Type of change

- Enhancement / new feature

### Description

This PR introduce new Selenium pod, which can be used for OLM or web testing (console for example). New test group called `olm` was added for OLM group of tests. Current tests cover only basic functionality without upgrade.

It still needs some check and refactor for those parts, which was copied from [EnMasse](https://github.com/EnMasseProject/enmasse) project.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass
- [ ] Update documentation
- [ ] Double check copied code

### CODE REVIEW

File: systemtest/src/main/java/io/strimzi/systemtest/Resources.java [hunk 1]
```diff
@@ -295,57 +299,57 @@ public KafkaBuilder defaultKafka(String name, int kafkaReplicas, int zookeeperRe
...
-                            .endJvmOptions()
-                        .endKafka()
-                        .withNewZookeeper()
-                            .withReplicas(zookeeperReplicas)
-                            .withResources(new ResourceRequirementsBuilder()
-                                .addToRequests("memory", new Quantity("1G")).build())
-                            .withMetrics(new HashMap<>())
-                            .withNewReadinessProbe()
+                .withMetadata(new ObjectMetaBuilder().withName(name).withNamespace(client().getNamespace()).build())
+                .withNewSpec()
```
Comment (stanlyDoge) [inline systemtest/src/main/java/io/strimzi/systemtest/Resources.java]: I know this is annoying (I suspect IDEA does this), but we should keep the formatted style.

File: systemtest/src/main/java/io/strimzi/systemtest/Resources.java [hunk 2]
```diff
@@ -375,15 +379,15 @@ DoneableKafkaConnect kafkaConnect(String name, int kafkaConnectReplicas) {
 
     private KafkaConnectBuilder defaultKafkaConnect(String name, int kafkaConnectReplicas) {
         return new KafkaConnectBuilder()
-            .withMetadata(new ObjectMetaBuilder().withName(name).withNamespace(client().getNamespace()).build())
-            .withNewSpec()
+                .withMetadata(new ObjectMetaBuilder().withName(name).withNamespace(client().getNamespace()).build())
```
Comment (stanlyDoge) [inline systemtest/src/main/java/io/strimzi/systemtest/Resources.java]: Formatting

File: systemtest/src/main/java/io/strimzi/systemtest/Resources.java [hunk 3]
```diff
@@ -419,15 +424,15 @@ DoneableKafkaConnectS2I kafkaConnectS2I(String name, int kafkaConnectS2IReplicas
 
     private KafkaConnectS2IBuilder defaultKafkaConnectS2I(String name, int kafkaConnectS2IReplicas, String kafkaClusterName) {
         return new KafkaConnectS2IBuilder()
-            .withMetadata(new ObjectMetaBuilder().withName(name).withNamespace(client().getNamespace()).build())
-            .withNewSpec()
+                .withMetadata(new ObjectMetaBuilder().withName(name).withNamespace(client().getNamespace()).build())
```
Comment (stanlyDoge) [inline systemtest/src/main/java/io/strimzi/systemtest/Resources.java]: Indentation

File: systemtest/src/main/java/io/strimzi/systemtest/Resources.java (hunk 2 above)
Comment (stanlyDoge) [inline systemtest/src/main/java/io/strimzi/systemtest/Resources.java]: Indentation

File: systemtest/src/main/java/io/strimzi/systemtest/Resources.java [hunk 4]
```diff
@@ -295,57 +299,57 @@ public KafkaBuilder defaultKafka(String name, int kafkaReplicas, int zookeeperRe
...
-                                .withGcLoggingEnabled(false)
-                            .endJvmOptions()
-                        .endKafka()
-                        .withNewZookeeper()
-                            .withReplicas(zookeeperReplicas)
-                            .withResources(new ResourceRequirementsBuilder()
-                                .addToRequests("memory", new Quantity("1G")).build())
-                            .withMetrics(new HashMap<>())
-                            .withNewReadinessProbe()
+                .withMetadata(new ObjectMetaBuilder().withName(name).withNamespace(client().getNamespace()).build())
```
Comment (stanlyDoge) [inline systemtest/src/main/java/io/strimzi/systemtest/Resources.java]: Indentation

File: systemtest/src/main/java/io/strimzi/systemtest/Resources.java [hunk 5]
```diff
@@ -276,12 +280,12 @@ public DoneableKafka kafkaEphemeral(String name, int kafkaReplicas, int zookeepe
     public DoneableKafka kafkaJBOD(String name, int kafkaReplicas, JbodStorage jbodStorage) {
         return kafka(defaultKafka(name, kafkaReplicas).
                 editSpec()
-                    .editKafka()
-                        .withStorage(jbodStorage)
-                    .endKafka()
-                    .editZookeeper().
+                .editKafka()
```
Comment (stanlyDoge) [inline systemtest/src/main/java/io/strimzi/systemtest/Resources.java]: Indentation

File: systemtest/src/main/java/io/strimzi/systemtest/Resources.java [hunk 6]
```diff
@@ -457,24 +462,24 @@ DoneableKafkaMirrorMaker kafkaMirrorMaker(String name, String sourceBootstrapSer
 
     private KafkaMirrorMakerBuilder defaultMirrorMaker(String name, String sourceBootstrapServer, String targetBootstrapServer, String groupId, int mirrorMakerReplicas, boolean tlsListener) {
         return new KafkaMirrorMakerBuilder()
-            .withMetadata(new ObjectMetaBuilder().withName(name).withNamespace(client().getNamespace()).build())
-            .withNewSpec()
+                .withMetadata(new ObjectMetaBuilder().withName(name).withNamespace(client().getNamespace()).build())
```
Comment (stanlyDoge) [inline systemtest/src/main/java/io/strimzi/systemtest/Resources.java]: Indentation

File: systemtest/src/main/java/io/strimzi/systemtest/Resources.java [hunk 7]
```diff
@@ -295,57 +299,57 @@ public KafkaBuilder defaultKafka(String name, int kafkaReplicas, int zookeeperRe
...
-                            .withNewJvmOptions()
-                                .withGcLoggingEnabled(false)
-                            .endJvmOptions()
-                        .endZookeeper()
-                        .withNewEntityOperator()
-                            .withNewTopicOperator().withImage(tOImage).endTopicOperator()
-                            .withNewUserOperator().withImage(uOImage).endUserOperator()
-                        .endEntityOperator()
-                    .endSpec();
+                .withNewEphemeralStorage().endEphemeralStorage()
```
Comment (stanlyDoge) [inline systemtest/src/main/java/io/strimzi/systemtest/Resources.java]: Indentation

File: systemtest/src/main/java/io/strimzi/systemtest/Resources.java [hunk 8]
```diff
@@ -457,24 +462,24 @@ DoneableKafkaMirrorMaker kafkaMirrorMaker(String name, String sourceBootstrapSer
...
+                .withBootstrapServers(tlsListener ? targetBootstrapServer + "-kafka-bootstrap:9093" : targetBootstrapServer + "-kafka-bootstrap:9092")
+                .addToConfig(ProducerConfig.ACKS_CONFIG, "all")
                 .endProducer()
                 .withResources(new ResourceRequirementsBuilder()
                         .addToRequests("memory", new Quantity("1G")).build())
                 .withMetrics(new HashMap<>())
-            .withReplicas(mirrorMakerReplicas)
-            .withWhitelist(".*")
-            .endSpec();
+                .withReplicas(mirrorMakerReplicas)
```
Comment (stanlyDoge) [inline systemtest/src/main/java/io/strimzi/systemtest/Resources.java]: Indentation

File: systemtest/src/main/java/io/strimzi/systemtest/Resources.java [hunk 9]
```diff
@@ -558,10 +563,10 @@ private void waitForDeletion(Kafka kafka) {
         LOGGER.info("Waiting when all the pods are terminated for Kafka {}", kafka.getMetadata().getName());
 
         IntStream.rangeClosed(0, kafka.getSpec().getZookeeper().getReplicas() - 1).forEach(podIndex ->
-            waitForPodDeletion(kafka.getMetadata().getName() + "-zookeeper-" + podIndex));
+                waitForPodDeletion(kafka.getMetadata().getName() + "-zookeeper-" + podIndex));
```
Comment (stanlyDoge) [inline systemtest/src/main/java/io/strimzi/systemtest/Resources.java]: Indentation

File: systemtest/src/main/java/io/strimzi/systemtest/Resources.java [hunk 10]
```diff
@@ -630,9 +635,9 @@ public DoneableKafkaTopic topic(String clusterName, String topicName, int partit
 
     public DoneableKafkaTopic topic(String clusterName, String topicName, int partitions, int replicas) {
         return topic(defaultTopic(clusterName, topicName, partitions, replicas)
-            .editSpec()
-            .addToConfig("min.insync.replicas", replicas)
-            .endSpec().build());
+                .editSpec()
```
Comment (stanlyDoge) [inline systemtest/src/main/java/io/strimzi/systemtest/Resources.java]: Indentation

File: systemtest/src/main/java/io/strimzi/systemtest/Resources.java [hunk 11]
```diff
@@ -558,10 +563,10 @@ private void waitForDeletion(Kafka kafka) {
         LOGGER.info("Waiting when all the pods are terminated for Kafka {}", kafka.getMetadata().getName());
 
         IntStream.rangeClosed(0, kafka.getSpec().getZookeeper().getReplicas() - 1).forEach(podIndex ->
-            waitForPodDeletion(kafka.getMetadata().getName() + "-zookeeper-" + podIndex));
+                waitForPodDeletion(kafka.getMetadata().getName() + "-zookeeper-" + podIndex));
 
         IntStream.rangeClosed(0, kafka.getSpec().getKafka().getReplicas() - 1).forEach(podIndex ->
-            waitForPodDeletion(kafka.getMetadata().getName() + "-kafka-" + podIndex));
+                waitForPodDeletion(kafka.getMetadata().getName() + "-kafka-" + podIndex));
```
Comment (stanlyDoge) [inline systemtest/src/main/java/io/strimzi/systemtest/Resources.java]: Indentation

File: systemtest/src/main/java/io/strimzi/systemtest/Resources.java [hunk 12]
```diff
@@ -996,14 +1002,14 @@ private PodSpec createClientSpec(boolean tlsListener, KafkaUser... kafkaUsers) {
                 .withName(Constants.KAFKA_CLIENTS)
                 .withImage(Environment.TEST_CLIENT_IMAGE)
                 .addNewPort()
-                    .withContainerPort(Environment.KAFKA_CLIENTS_DEFAULT_PORT)
+                .withContainerPort(Environment.KAFKA_CLIENTS_DEFAULT_PORT)
```
Comment (stanlyDoge) [inline systemtest/src/main/java/io/strimzi/systemtest/Resources.java]: Indentation

File: systemtest/src/main/java/io/strimzi/systemtest/Resources.java [hunk 13]
```diff
@@ -897,50 +902,51 @@ DoneableClusterRoleBinding clusterRoleBinding(ClusterRoleBinding clusterRoleBind
...
+                .endSpec()
+                .build();
 
         return createNewDeployment(kafkaClient, namespace);
     }
 
     public static ServiceBuilder getSystemtestsServiceResource(String appName, int port, String namespace) {
         return new ServiceBuilder()
-            .withNewMetadata()
+                .withNewMetadata()
```
Comment (stanlyDoge) [inline systemtest/src/main/java/io/strimzi/systemtest/Resources.java]: Indentation

File: systemtest/src/main/java/io/strimzi/systemtest/Resources.java [hunk 14]
```diff
@@ -897,50 +902,51 @@ DoneableClusterRoleBinding clusterRoleBinding(ClusterRoleBinding clusterRoleBind
...
-    DoneableDeployment deployKafkaClients(boolean tlsListener, String clusterName, String namespace) {
+    public DoneableDeployment deployKafkaClients(boolean tlsListener, String clusterName, String namespace) {
         return deployKafkaClients(tlsListener, clusterName, namespace, null);
     }
-    DoneableDeployment deployKafkaClients(boolean tlsListener, String clusterName, String namespace, KafkaUser... kafkaUsers) {
+
+    public DoneableDeployment deployKafkaClients(boolean tlsListener, String clusterName, String namespace, KafkaUser... kafkaUsers) {
         Deployment kafkaClient = new DeploymentBuilder()
-            .withNewMetadata()
+                .withNewMetadata()
```
Comment (stanlyDoge) [inline systemtest/src/main/java/io/strimzi/systemtest/Resources.java]: Indentation

File: systemtest/src/main/java/io/strimzi/systemtest/Resources.java [hunk 15]
```diff
@@ -832,60 +837,60 @@ private ClusterRoleBindingBuilder defaultClusterRoleBinding(String yamlPath, Str
...
-                    .withName("strimzi-topic-operator")
-                .endRoleRef()
-                .withSubjects(new SubjectBuilder()
-                    .withKind("ServiceAccount")
-                    .withName("strimzi-cluster-operator")
-                    .withNamespace(namespace)
-                    .build()
-                )
-                .build()
+                new ClusterRoleBindingBuilder()
```
Comment (stanlyDoge) [inline systemtest/src/main/java/io/strimzi/systemtest/Resources.java]: Indentation

File: systemtest/src/main/java/io/strimzi/systemtest/Resources.java [hunk 16]
```diff
@@ -832,60 +837,60 @@ private ClusterRoleBindingBuilder defaultClusterRoleBinding(String yamlPath, Str
...
-                    .withName("strimzi-entity-operator")
-                .endRoleRef()
-                .withSubjects(new SubjectBuilder()
-                    .withKind("ServiceAccount")
-                    .withName("strimzi-cluster-operator")
-                    .withNamespace(namespace)
-                    .build()
-                )
-                .build()
+                new ClusterRoleBindingBuilder()
```
Comment (stanlyDoge) [inline systemtest/src/main/java/io/strimzi/systemtest/Resources.java]: Indentation

File: systemtest/src/main/java/io/strimzi/systemtest/Resources.java [hunk 17]
```diff
@@ -832,60 +837,60 @@ private ClusterRoleBindingBuilder defaultClusterRoleBinding(String yamlPath, Str
...
-                    .withName("strimzi-cluster-operator-namespaced")
-                .endRoleRef()
-                .withSubjects(new SubjectBuilder()
-                    .withKind("ServiceAccount")
-                    .withName("strimzi-cluster-operator")
-                    .withNamespace(namespace)
-                    .build()
-                )
-                .build()
+                new ClusterRoleBindingBuilder()
```
Comment (stanlyDoge) [inline systemtest/src/main/java/io/strimzi/systemtest/Resources.java]: Indentation

File: systemtest/src/main/java/io/strimzi/systemtest/Resources.java [hunk 18]
```diff
@@ -801,7 +806,7 @@ private RoleBindingBuilder defaultRoleBinding(String yamlPath, String namespace)
         return new RoleBindingBuilder(getRoleBindingFromYaml(yamlPath))
                 .withApiVersion("rbac.authorization.k8s.io/v1")
                 .editFirstSubject()
-                    .withNamespace(namespace)
+                .withNamespace(namespace)
```
Comment (stanlyDoge) [inline systemtest/src/main/java/io/strimzi/systemtest/Resources.java]: Indentation

File: systemtest/src/main/java/io/strimzi/systemtest/Resources.java [hunk 19]
```diff
@@ -822,7 +827,7 @@ private ClusterRoleBindingBuilder defaultClusterRoleBinding(String yamlPath, Str
         return new ClusterRoleBindingBuilder(getClusterRoleBindingFromYaml(yamlPath))
                 .withApiVersion("rbac.authorization.k8s.io/v1")
                 .editFirstSubject()
-                    .withNamespace(namespace)
+                .withNamespace(namespace)
```
Comment (stanlyDoge) [inline systemtest/src/main/java/io/strimzi/systemtest/Resources.java]: Indentation

File: systemtest/src/main/java/io/strimzi/systemtest/Resources.java [hunk 20]
```diff
@@ -749,16 +754,16 @@ private DeploymentBuilder defaultCLusterOperator(String namespace, String operat
...
-                        .addToMatchLabels("name", Constants.STRIMZI_DEPLOYMENT_NAME)
-                    .endSelector()
-                    .editTemplate()
-                        .editSpec()
-                            .editFirstContainer()
-                                .withImage(StUtils.changeOrgAndTag(coImage))
-                            .endContainer()
-                        .endSpec()
-                    .endTemplate()
+                .withNewSelector()
```
Comment (stanlyDoge) [inline systemtest/src/main/java/io/strimzi/systemtest/Resources.java]: Indentation

File: systemtest/src/main/java/io/strimzi/systemtest/Resources.java [hunk 21]
```diff
@@ -682,8 +687,8 @@ public DoneableKafkaUser scramShaUser(String clusterName, String name) {
                         .addToLabels("strimzi.io/cluster", clusterName)
                         .build())
                 .withNewSpec()
-                    .withNewKafkaUserScramSha512ClientAuthentication()
-                    .endKafkaUserScramSha512ClientAuthentication()
+                .withNewKafkaUserScramSha512ClientAuthentication()
```
Comment (stanlyDoge) [inline systemtest/src/main/java/io/strimzi/systemtest/Resources.java]: Indentation

File: systemtest/src/main/java/io/strimzi/systemtest/Resources.java [hunk 22]
```diff
@@ -667,8 +672,8 @@ public DoneableKafkaUser tlsUser(String clusterName, String name) {
                         .addToLabels("strimzi.io/cluster", clusterName)
                         .build())
                 .withNewSpec()
-                    .withNewKafkaUserTlsClientAuthentication()
-                    .endKafkaUserTlsClientAuthentication()
+                .withNewKafkaUserTlsClientAuthentication()
```
Comment (stanlyDoge) [inline systemtest/src/main/java/io/strimzi/systemtest/Resources.java]: Indentation

File: systemtest/src/main/java/io/strimzi/systemtest/Resources.java [hunk 23]
```diff
@@ -643,10 +648,10 @@ private KafkaTopicBuilder defaultTopic(String clusterName, String topicName, int
                                 .withName(topicName)
                                 .withNamespace(client().getNamespace())
                                 .addToLabels("strimzi.io/cluster", clusterName)
-                .build())
+                                .build())
                 .withNewSpec()
-                    .withPartitions(partitions)
-                    .withReplicas(replicas)
+                .withPartitions(partitions)
```
Comment (stanlyDoge) [inline systemtest/src/main/java/io/strimzi/systemtest/Resources.java]: Indentation

**Comment by tombentley** [comment]
I can see we have tests for OpenShift and OpenShift4, but OLM is not Kubernetes-specific, so shouldn't we have tests on normal Kubernetes?

**Comment by Frawless** [comment]
@tombentley well this PR contains only tests. It assumes, that cluster, on which the tests are being executed, has configured OLM with manifests which you want. Cluster preparations should be done separately, for example in jenkins pipeline (as we have it now), cause it contains several steps, which will be very ugly in java code.

**Comment by Frawless** [comment]
After discussion, we decided to close this PR and new one with different approach for OLM tests.

**Pull request closed without merging**